### DISCUSSION TITLE
Script Editor Methods list to organize under Code Regions

### DISCUSSION THREAD

**Discussion by mikemike37** [godotengine/godot-proposals#11137] [Editor]
**Describe the project you are working on**
A 2D/3D roguelike using GDScript extensively

**Describe the problem or limitation you are having in your project**
I have large scripts with code regions and many functions

**Describe the feature / enhancement and how it helps to overcome the problem or limitation**
Prompted by this Reddit thread:
https://www.reddit.com/r/godot/comments/1gl18lx/how_bad_is_this_workaround_is_there_an_elegant/
...it would be nice if code regions were shown in the method list, ideally as collapsible, like a tree view. This appeared to be a popular proposal, based on reddit upvotes.

**Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams**
![image](https://github.com/user-attachments/assets/2610e036-dc64-4904-afb8-8f3ad50a8c55)

**If this enhancement will not be used often, can it be worked around with a few lines of script?**
You can see from that reddit thread what somebody was doing to try and organize the method list.

**Is there a reason why this should be core and not an add-on in the asset library?**
I expect every user will benefit from this lightweight change to a very core feature (the script editor)

**Comment by tetrapod00** [comment]
There's a proposal for this: https://github.com/godotengine/godot-proposals/issues/8403, though the proposed UI in that issue is a little different than your mockup.

Feel free to comment on that proposal with any input you have, such as alternate mockups

**Reply by mikemike37** [reply]
ah thanks! I really tried to look see if a proposal for this existed, but didn't spot that. Happy to close this as a dupe. I've added my mockup over there.

EDIT: ah looks like that proposal came a day or two after this. It has better diagrams, so closing this one as dupe.

**Comment by sockeye-d** [comment]
I like this, but the text for the region should be grayed out or something to show that it's not selectable
